src/pages/inventory/index.tsx

```diff
@@ -0,0 +1,265 @@
+  const [type, setType] = useState('');
+  const [category, setCategory] = useState('');
+  const [status, setStatus] = useState('');
+  const [search, setSearch] = useState('');
+  const [anchorType, setAnchorType] = useState<null | HTMLElement>(null);
+  const [anchorCategory, setAnchorCategory] = useState<null | HTMLElement>(null);
+  const [anchorStatus, setAnchorStatus] = useState<null | HTMLElement>(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const itemsPerPage = 10;
+
+  const indexOfLastItem = currentPage * itemsPerPage;
+  const indexOfFirstItem = indexOfLastItem - itemsPerPage;
+  const currentItems = data.slice(indexOfFirstItem, indexOfLastItem);
+
+  const handleTypeClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorType(event.currentTarget);
+  };
+  const handleCategoryClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorCategory(event.currentTarget);
+  };
+  const handleStatusClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorStatus(event.currentTarget);
+  };
+
+  const handleTypeClose = () => {
+    setAnchorType(null);
+  }
+
+  const handleMenuTypeClick = (value: string) => {
+    setType(value)
+    handleFilter();
+    handleTypeClose();
+  }
+
+  const handleCategoryClose = () => {
+    setAnchorCategory(null);
+  }
+
+  const handleMenuCategoryClick = (value: string) => {
+    setCategory(value);
+    handleFilter();
+    handleCategoryClose();
+  }
+
+  const handleStatusClose = () => {
+    setAnchorStatus(null);
+  }
+
+  const handleMenuStatusClick = (value: string) => {
+    setStatus(value);
+    handleFilter();
+    handleStatusClose();
+  }
+
+  const clearTypeFilter = () => {
+    setType('');
+  }
+  const clearCategoryFilter = () => {
+    setCategory('');
+  }
+  const clearStatusFilter = () => {
+    setStatus('');
+  }
+
+  const handleSearch = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearch(event.target.value)
+  }
+
+  const handlePageChange = (_event: React.ChangeEvent<unknown>, value: number) => {
+    setCurrentPage(value);
+  }
+
+  const handleFilter = () => {
+    const searchFiltered = originalData.filter((row: { item: string, type: string, category: string, quantity: number; }) => {
+
+      const matchesType = type ? row.type.toLowerCase().includes(type.toLowerCase()) : true;
+
+      const matchesCategory = category ? row.category.toLowerCase().includes(category.toLowerCase()) : true;
+
+      // const matchesStatus = status ? item.status.toLowerCase().includes(status.toLowerCase()) : true; //
+
+      const matchesSearch = search
+        ? row.item.toLowerCase().includes(search.toLowerCase()) ||
+        row.type.toLowerCase().includes(search.toLowerCase()) ||
+        row.category.toLowerCase().includes(search.toLowerCase()) ||
+        // item.status.toLowerCase().includes(search.toLowerCase()) ||
+        row.quantity.toString().toLowerCase().includes(search.toLowerCase())
+        : true;
```

Comment:
By storing lowercase search term in a variable before the comparisons, can reduce calling toLowerCase() multiple times.

```suggestion
      const lowerCaseSearch = search.toLowerCase();
      const matchesSearch = search
        ? row.item.toLowerCase().includes(lowerCaseSearch) ||
        row.type.toLowerCase().includes(lowerCaseSearch) ||
        row.category.toLowerCase().includes(lowerCaseSearch) ||
        // item.status.toLowerCase().includes(lowerCaseSearch) ||
        row.quantity.toString().toLowerCase().includes(lowerCaseSearch)
        : true;
```

Reply:
Updated